R/exposure_metrics.R

```diff
@@ -16,17 +16,22 @@ NULL
 #'
 #' @param sim_output output of a `PKPDsim::sim` call
 #' @param auc_comp auc compartment
+#' @param target_time target time from `get_sampling_times_from_scheme`
+#' @param target_type type of AUC target in `c(auc24, auc12, cum_auc)`.
 #' @returns `calc_auc_from_sim` returns a numeric vector of AUCs between each
 #'   simulated time point. Control time period over which AUC should be
 #'   calculated using `target_time` argument to `PKPDsim::sim`.
 #' @export
 
-calc_auc_from_sim <- function(sim_output, auc_comp) {
-  aucs <- sim_output$y[sim_output$comp == auc_comp]
-  if (length(aucs) == 1) {
-    aucs # i.e., cumulative
-  } else {
-    diff(aucs)
+calc_auc_from_sim <- function(sim_output, auc_comp, target_time, target_type) {
```

Comment:
The original version of these two functions will not work if target type if AUC24 or AUC12 and there are multiple times requested.

For example:

```
target_time <- c(200, 300)
target_type <- "auc24"
```

Will result in a sim_output that looks like:

```
sim_output <- data.frame(
  t = c(174, 200, 274, 300),
  y = c(1, 2, 3, 4),
  comp = 15 # assume AUC comp
)
```

which would then return a list of 3 AUCs using the original function.